turbopack/crates/turbo-tasks/src/raw_vc.rs

```diff
@@ -58,6 +58,7 @@ impl Display for CellId {
 pub enum RawVc {
     TaskOutput(TaskId),
     TaskCell(TaskId, CellId),
+    LocalOutput(TaskId, LocalTaskId),
     #[serde(skip)]
     LocalCell(ExecutionId, LocalCellId),
```

Comment:
That looks a bit inconsitent/weird to me. Why does a LocalOutput need a TaskId and a LocalCell and ExecutionId? A LocalOutput shouldn't escape the current task (similar to LocalCell), so it shouldn't need that.

Reply:
In theory, since a `LocalOutput` shouldn't be able to escape the current task, this *shouldn't* be needed, but:

- The `ResolvedValue` trait is not infallible, even once it's used everywhere, as there will still be ways to store `Vc`s in global state. Storing `TaskId` here gives us a way to perform additional runtime checks (eventually maybe only in debug mode?).

- Making this check airtight (removing the need for a runtime check) would require adding lifetimes to `Vc` to prevent it from being stored globally, but we decided against that for now.

- These runtime checks are particularly useful in the interim before we're fully migrated to `ResolvedValue` and `ResolvedVc`.

- The short-term plan of storing `LocalOutput` globally to mitigate these escapes and allow us to roll out something sooner requires a `TaskId`: https://www.notion.so/vercel/RawVc-LocalOutput-aede5f463f594ca58396eb3fdaffd865?pvs=4#02af94e3b9b64a3690f3e467abc0a47e

This serves a role similar to `ExecutionId` on `LocalCell`, except that the short-term plan of storing `LocalOutput` globally sadly necessitates using a `TaskId` instead of an `ExecutionId` for now.